### PR TITLE
Use AATweather #444

### PR DESCRIPTION
**Pull request by fergusL**
- utility function to grab AAT weather 
- over-rides the pocs `is_weather_safe()` method so that it takes the result of the parent method and combines it with the AAT rain flag to determine if weather is safe

Potentially we might want to actually store the AAT data in a db somewhere, in which case I could make a script to query/store the data and add a new service to the docker-compose file to run this script.

Also looks like this change is going to break a bunch of tests that assume inserting a "safe" weather reading into the db will produce a corresponding safe result from the safety checks. Need a fixture or aat weather simulator

### CODE REVIEW

**Comment by fergusL** [comment]
Added something so tests wouldn't fail. Tests passing locally but github actions is failing because it can't pull the pocs base image

**Comment by wtgee** [comment]
@fergusL I'll take a closer look at this later, but we shouldn't be having any kind of `if` for tests in the actual code. If the db says `false` for safety then the weather is bad.

So you'll basically have to go the route of inserting into the database, which is what POCs is expecting. But it should be a simple `self.db.insert_current`. 

I'll look more later

File: src/huntsman/pocs/core.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def after_state(self, event_data):
         """
         self.say(f"Finished with the {self.state} state. The next state is {self.next_state}.")
 
-    # Private methods
+    # private methods
```
Comment (danjampro) [inline src/huntsman/pocs/core.py]: Just curious as to why this change? I have been commenting with capitals. Not important.

**Review by danjampro — COMMENTED** [review]
@fergusL looks good. The thing that is missing seems to be how the state machine uses the safety data. I think all we need is a logical `and` to check whether weather is safe, e.g. something like:

`is_safe = aat_is_open & super().is_safe()`

in `core.HuntsmanPOCS`. It might also be good to add a config option to ignore AAT weather if we need to.

File: src/huntsman/pocs/core.py (hunk 1 above)
Comment (fergusL) [reply src/huntsman/pocs/core.py]: umm not sure, probably some auto formatting from my editor. Definitely not an intentional change

**Review by danjampro — APPROVED** [review]
Looks good! Do we need to keep the read_aat_weather script? If so then it should probably import things from `safety.py`. If not then it can be deleted.
